File: writersblock-front/src/services/PlaceService.js
import axios from "axios";

const PLACE_API_BASE_URL = "http://localhost:8080/api/v1/place";

class PlaceService {
  savePlace(place) {
    return axios.post(PLACE_API_BASE_URL, place);
  }

  getPlaces() {
    return axios.get(PLACE_API_BASE_URL + "s");
  }

  getPlaceById(id){
    return axios.get(PLACE_API_BASE_URL + "/" + id);
  }

  getPlacesByName(name){
    return axios.get(PLACE_API_BASE_URL + "s/" + name);
  }

  deletePlace(id) {
    return axios.delete(PLACE_API_BASE_URL + "/" + id);
  }

  updatePlace(id, place){
    return axios.put(PLACE_API_BASE_URL + "/" + id, place);
  }
}

export default new PlaceService();

File: writersblock-back/src/main/java/com/muratoni/writersblock/service/PlaceService.java
package com.muratoni.writersblock.service;

import java.util.List;

import com.muratoni.writersblock.model.Place;

public interface PlaceService {

    Place createPlace(Place place);

    List<Place> getAllPlaces();

    Place getPlaceById(Long id);

    List<Place> getPlacesByName(String name);

    boolean deletePlace(Long id);

    Place updatePlace(Long id, Place place);
    
}
